python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -0,0 +1,921 @@
+
+        for pos_neg in ['pos', 'neg']:
+
+            key = schema.addField(
+                schema.join(name, pos_neg, "flux"), type=float, units="dn",
+                doc="Dipole {0} lobe flux".format(pos_neg))
+            setattr(self, ''.join((pos_neg, 'FluxKey')), key)
+
+            key = schema.addField(
+                schema.join(name, pos_neg, "fluxSigma"), type=float, units="pixels",
+                doc="1-sigma uncertainty for {0} dipole flux".format(pos_neg))
+            setattr(self, ''.join((pos_neg, 'FluxSigmaKey')), key)
+
+            for x_y in ['x', 'y']:
+                key = schema.addField(
+                    schema.join(name, pos_neg, "centroid", x_y), type=float, units="pixels",
+                    doc="Dipole {0} lobe centroid".format(pos_neg))
+                setattr(self, ''.join((pos_neg, 'CentroidKey', x_y.upper())), key)
+
+        for x_y in ['x', 'y']:
+            key = schema.addField(
+                schema.join(name, "centroid", x_y), type=float, units="pixels",
+                doc="Dipole centroid")
+            setattr(self, ''.join(('centroidKey', x_y.upper())), key)
+
+        self.fluxKey = schema.addField(
+            schema.join(name, "flux"), type=float, units="dn",
+            doc="Dipole overall flux")
+
+        self.orientationKey = schema.addField(
+            schema.join(name, "orientation"), type=float, units="deg",
+            doc="Dipole orientation")
+
+        self.separationKey = schema.addField(
+            schema.join(name, "separation"), type=float, units="pixels",
+            doc="Pixel separation between positive and negative lobes of dipole")
+
+        self.chi2dofKey = schema.addField(
+            schema.join(name, "chi2dof"), type=float,
+            doc="Chi2 per degree of freedom of dipole fit")
+
+        self.signalToNoiseKey = schema.addField(
+            schema.join(name, "signalToNoise"), type=float,
+            doc="Estimated signal-to-noise of dipole fit")
+
+        self.classificationFlagKey = schema.addField(
+            schema.join(name, "flag", "classification"), type="Flag",
+            doc="Flag indicating diaSource is classified as a dipole")
+
+        self.flagKey = schema.addField(
+            schema.join(name, "flag"), type="Flag",
+            doc="General failure flag for dipole fit")
+
+        self.edgeFlagKey = schema.addField(
+            schema.join(name, "flag", "edge"), type="Flag",
+            doc="Flag set when dipole is too close to edge of image")
+
+    def measure(self, measRecord, exposure, posImage=None, negImage=None):
+        """Perform the non-linear least squares minimization.
+
+        The main functionality of this routine was placed outside of
+        this plugin (into `DipoleFitAlgorithm.fitDipole()`) so that
+        `DipoleFitAlgorithm.fitDipole()` can be called separately for
+        testing (see `tests/testDipoleFitter.py`)
+        """
```

Comment:
I like this idea. I will do the latter option. I like having the ability to run the algorithm separately for testing. This seems to require me to declare `DipoleFitAlgorithm` **prior** to declaring `DipoleFitTask`.